### PR TITLE
[doc] Documentation for SE3 with examples and nice figures for Joints.

### PR DESCRIPTION
**Pull request by tdouin**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ Xcode*
 *~
 *.pyc
 coverage*
+.vscode*
+_build*
```
Comment (jcarpent) [inline .gitignore]: This line is specific to your settings. I would prefer to not put it here. @tdouin Do you agree?

Comment (tdouin) [reply .gitignore]: yes sure.

Comment (nim65s) [reply .gitignore]: @tdouin: for information, you can add eg.
```
[core]
    excludesfile = ~/.global_gitignore
```
In your `~/.gitconfig`, and put those `.vscode*` & `_build*` in your `~/.global_gitignore`, and never worry about those again :)

**Comment by jcarpent** [comment]
@tdouin I'm making some modifications on your branch. These are just minor improvements in your excellent text to make it a bit clearer. I hope you won't mine.

File: .gitignore (hunk 1 above)
Comment (tdouin) [reply .gitignore]: Ok i get it, thanks for the advice

**Comment by jmirabel** [comment]
@tdouin nice work !

**Comment by jcarpent** [comment]
> May I ask why? Some people like to call their build folders _build-something. Not me, but others do. Why removing this possibility?

We cannot handle all the cases, and at this stage, the common way to do it is to cover the classic cases with Visual Studio, etc. The rest is very dependent on the user setup.

**Comment by jcarpent** [comment]
@jmirabel Do you want to add other changes or comments?

**Comment by jmirabel** [comment]
> @jmirabel Do you want to add other changes or comments?

No

**Comment by jcarpent** [comment]
Very nice work @tdouin. Thanks a lot!

**Pull request merged**